File: BankManagementSolution/GoBankManagement/model/Customer.go
package model

import "gorm.io/gorm"

type AccountType int

const (
	Savings      AccountType = iota //0
	Checking                        //1
	FixedDeposit                    //2
)

type CustomerStatus int

const (
	Activated   CustomerStatus = iota //0
	Deactivated                       //1
)

type Customer struct {
	gorm.Model
	CustomerID    int            `gorm:"primaryKey"`
	AccountNumber string         `gorm:"not null;unique"`
	FirstName     string         `gorm:"not null"`
	LastName      string         `gorm:"not null"`
	Email         string         `gorm:"not null;unique"`
	PhoneNumber   string         `gorm:"not null"`
	Address       string         `gorm:"not null"`
	AccountType   AccountType    `gorm:"not null"`
	CStatus       CustomerStatus `gorm:"default:0"`
	
}

File: BankManagementSolution/BankManagementApplication/Models/Customer.cs
using System.ComponentModel.DataAnnotations;

namespace BankManagementApplication.Models
{
    public enum CustomerStatus
    {
        Activated = 0, Deactivated = 1
    }
    public enum AccountType
    {
        Savings = 0,
        Checking = 1,
        FixedDeposit = 2
    }

    public class Customer
    {
        [Key]
        public int CustomerId { get; set; }

        [Required]
        public string AccountNumber { get; set; }

        [Required]
        public string FirstName { get; set; }

        [Required]
        public string LastName { get; set; }


        [Required]
        public string Email { get; set; }

        [Required]
        public string PhoneNumber { get; set; }

        [Required]
        public string Address { get; set; }

        public AccountType AccountType { get; set; }
        public CustomerStatus CStatus { get; set; } = CustomerStatus.Activated;

    }
}
